File: 0409 longest-palindrome/solution.go
func longestPalindrome(s string) int {
    var arr [128]int
    for _,v := range s {
        arr[v]++
    }
    sum:=0
    b:=false
    for _,v := range arr {
        if(v%2==0) {
            sum+=v
        } else {
            b=true
            sum+=v-1
        }
    }
    if(b) {
        sum++
    }
    return sum
}

File: 0409 longest-palindrome/solution.rs
impl Solution {
    pub fn longest_palindrome(s: String) -> i32 {
        let mut f = vec![0;126];
        for c in s.chars() {
            f[c as usize]+=1;
        }
        let mut sum =0;
        let mut b = false;
        for i in f {
            if i!=0 && (i&1)==0 {
                sum+=i;
            } else if i!=0 {
                sum+=i-1;
                b = true;
            }
        }
        if b {
            sum+=1;
        }
        sum
    }
}
